src/parser.rs

```diff
@@ -0,0 +1,1120 @@
+use ast::*;
+use std::collections::HashMap;
+use token::Token;
+use lexer::Lexer;
+
+type ParseError = String;
+type ParseErrors = Vec<ParseError>;
+pub type ParseResult<T> = Result<T, ParseError>;
```

Comment:
The error handling examples in both The Rust Programming Language and Programming Rust used just `Return` as a name. But when I tried to do that I got some weird error about cycles. Is that not allowed anymore and is it now the convention to do `<Thing>Result` and `<Thing>Error`?

Reply:
I assume you mean that you tried `pub type Result<T> = Result<T, ParseError>;`? This doesn't work because the compiler thinks you are defining the type alias in terms of itself, hence the `cycle detected` error (probably not the best error message).

Try using the fully qualified name instead:

```rust
pub type Result<T> = ::std::result::Result<T, ParseError>;
```